rust/rope/src/diff.rs

```diff
@@ -192,8 +208,23 @@ fn longest_increasing_region_set(items: &[(usize, usize)]) -> Vec<(usize, usize)
 }
 
 #[inline]
-fn non_ws_offset(s: &str) -> usize {
-    s.as_bytes().iter().take_while(|b| **b == b' ' || **b == b'\t').count()
+fn non_ws_offset(s: &RopeSlice) -> usize {
+    let mut cursor = Cursor::new(s.rope, s.range.start);
+
+    let mut result = 0;
+    let mut looping = true;
+    while looping {
+        if let Some(ch) = cursor.next_codepoint() {
+            if ch == ' ' || ch == '\t' {
+                result += 1;
+            } else {
+                looping = false;
+            }
+        } else {
+            looping = false;
```

Comment:
`break` would do the job.

Reply:
Fixed. This loop also used `while let` now.